src/devtools/client/debugger/src/reducers/pause.ts

```diff
@@ -171,6 +182,25 @@ const pauseSlice = createSlice({
       state.selectedFrameId = frame ? { pauseId: frame.pauseId, frameId: frame.protocolId } : null;
       state.threadcx.pauseCounter++;
       state.pausePreviewLocation = null;
+      if (
+        time &&
+        state.pauseHistoryIndex === state.pauseHistory.length - 1 &&
+        state.pauseHistory[state.pauseHistory.length - 1]?.time !== time
+      ) {
```

Comment:
Why are we only inserting pauses if we're at the end of the pause history? Maybe it's for a good reason but it's not obvious to me. Shouldn't we insert them regardless?

Reply:
Yep makes sense to also include other pauses, thank you for pointing it out. I will make the appropritate changes.

Reply:
I think this should work like a browser history does: all entries in the `pauseHistory` after `pauseHistoryIndex` should be removed and then the new pause added.

Reply:
I believe, it works the way as u have mentioned. I am removing the `pauseHistory` after the `pauseHistoryIndex`.